flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/FunctionITCase.java

```diff
@@ -237,6 +239,16 @@ public void testCreateTemporarySystemFunctionByUsingJar() {
         assertThat(Arrays.asList(tEnv().listFunctions())).doesNotContain("f10");
     }
 
+    @Test
+    public void testCreateTemporarySystemFunctionWithTableAPI() {
```

Comment:
Please refer to the `testUserDefinedTemporarySystemFunctionByUsingJar`, also add some tests that use UDF registered by table api.